src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -97,17 +97,15 @@ def _get_cloud_console_token_endpoint():
 # pylint: disable=too-many-lines,too-many-instance-attributes,unused-argument
 class Profile(object):
 
-    def __init__(self, storage=None, auth_ctx_factory=None, use_global_creds_cache=True,
+    def __init__(self, authority=None, storage=None, auth_ctx_factory=None, use_global_creds_cache=True,
```

Comment:
>authority=None, [](start = 23, length = 15)

it seems tricky here to pass your authority instead of use cli_ctx, could you please configure self.cli_ctx.cloud.endpoints.active_directory in your CLI code?

Reply:
Great idea, I have changed the code, thank you.

I might keep the `self._authority` though because there is the `.replace('https://', ''` part of the authority creation:
https://github.com/Azure/azure-cli/blob/b6c8e436888f93bee8a1e4b45700671e754b8500/src/azure-cli-core/azure/cli/core/_profile.py#L107

I think it is safer if it is done only once.

What do you think?